src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceWorker.cs

```diff
@@ -250,6 +250,7 @@ private Dictionary<string, DatabaseStatusReport> CollectDatabaseInformation(Tran
                 {
                     var now = dbInstance.Time.GetUtcNow();
                     FillReplicationInfo(dbInstance, report);
+                    report.LastClusterWideTransactionRaftIndex = dbInstance.ClusterWideTransactionIndexWaiter.LastIndex;
```

Comment:
I don't like the fact that they are separated.
Would it be okay to simply `FillClusterTransactionInfo` here?

Reply:
By "they," do you mean LastCompletedClusterTransaction and ClusterWideTransactionIndexWaiter?
Separating LastTransactionId makes sense, right?

The issue I found with moving LastCompletedClusterTransaction as well is that it could cause confusion for future readers of this code.
The assumption might be that if this value is also moved out of the "NoChange" section, it could change even when the storage remains unchanged, which is not true.

I moved ClusterWideTransactionIndexWaiter so it would be available later when ClusterMaintenanceSupervisor handles an "unchanged" report.
If I move LastCompletedClusterTransaction in the ClusterMaintenanceWorker, should I also handle that value in the ClusterMaintenanceSupervisor, as I did for ClusterWideTransactionIndexWaiter?

If I don’t move LastCompletedClusterTransaction, it will be unused (it is anyway unchanged but it will also be unused).
If I do move it, it might confuse future reviewers into thinking this value could change even when DocumentStorage is marked as "NoChange."

I prefer to keep it separate to make it clear that LastCompletedClusterTransaction does not change when the storage is unchanged—both in ClusterMaintenanceSupervisor and ClusterMaintenanceWorker.

But if you prefer to put them together I will do that.

Reply:
done